src/Connection/Internal/Http2ConnectionProcessor.php

```diff
@@ -594,7 +594,7 @@ static function () {
             $onPush = $stream->request->getPushHandler();
 
             try {
-                yield call($onPush, $stream->request, $stream->pendingResponse->promise());
+                yield call($onPush, $stream->request, $stream->pendingResponse->promise(), $tokenSource);
```

Comment:
Because of the `finally` below it seems like it's only possible to cancel while the promise returned from `$onPush` has not resolved, so it doesn't seem to make anything easier.

Reply:
D'oh, yes, that will have to be pushed to the code resolving the promise.

Reply:
Maybe we should just keep the current behavior?